base/reflection.jl

```diff
@@ -132,6 +143,9 @@ julia> fieldname(Rational, 2)
 ```
 """
 function fieldname(t::DataType, i::Integer)
+    if t.abstract
+        error("type does not have definite field names")
```

Comment:
Perhaps use `throw(ArgumentError("..."))`?

Reply:
`fieldcount` uses `error` for a similar case. Meanwhile, this function also uses `ArgumentError` for what is arguably a `BoundsError`. I guess I can change it though.

Reply:
Up to you, just figured I'd suggest it since typed errors can be handy and informative.